zerver/views/streams.py

```diff
@@ -395,10 +391,6 @@ def get_streams_backend(request, user_profile,
                              include_all_active)
     return json_success({"streams": streams})
 
-def get_public_streams_backend(request, user_profile):
-    return get_streams_backend(request, user_profile, include_public=True,
-                               include_subscribed=False, include_all_active=False)
```

Comment:
You will need to change your replacement `/json/streams` fetch to pass the arguments to only send public streams to resolve this issue so that the frontend gets the same behavior it had before.